corehq/pillows/app_submission_tracker.py

```diff
@@ -126,3 +131,95 @@ def get_sql_app_form_submission_tracker_reindexer():
         FormReindexAccessor(include_attachments=False)
     )
     return AppFormSubmissionReindexer(doc_provider, FORM_SQL, 'form_processor_xforminstancesql')
+
+
+@quickcache(['domain'], timeout=60 * 60)
+def _get_apps_for_domain(domain):
+    project = Domain.get_by_name(domain)
+    if project:
+        return [app._id for app in project.applications()]
+    else:
+        return []
+
+
+class UserAppFormSubmissionDocProcessor(BaseDocProcessor):
+    def __init__(self, pillow_processor):
+        self.pillow_processor = pillow_processor
+
+    def process_doc(self, doc):
+        form_submission_changes = self._doc_to_changes(doc)
+        for change in form_submission_changes:
+            try:
+                self.pillow_processor.process_change(None, change)
+            except Exception as e:
+                print(e)
+                return False
+        return True
+
+    def handle_skip(self, doc):
+        print('Unable to process user {}'.format(
+            doc['_id'],
+        ))
+        return True
+
+    def _doc_to_changes(self, doc):
+        changes = []
+        forms = []
+        apps = []
+        if doc['doc_type'] == 'CommCareUser':
+            apps = _get_apps_for_domain(doc['domain'])
+        else:
+            for domain in (dm['domain'] for dm in doc['domain_memberships']):
+                apps = _get_apps_for_domain(domain)
+        for app_id in apps:
+            query = FormES().app(app_id).user_id(doc['_id']).sort('received_on', desc=True).size(1)
+            hits = query.run().hits
+            if hits:
+                forms.append(hits[0])
```

Comment:
seems very similar to `get_last_form_submission` function, could possibly extend that. also nit should be in a separate esaccessor file